ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-text-input/activityTextInput.component.html

```diff
@@ -1,66 +1,75 @@
-        ></span>
-      </mat-option>
-    </mat-autocomplete>
-  </mat-form-field>
+        <mat-label *ngIf="block.label" [innerHTML]="block.label"></mat-label>
 
-  <ng-container *ngIf="block.confirmEntry">
-    <p
-      *ngIf="block.confirmPrompt"
-      class="ddp-question-prompt ddp-question-prompt-confirmation"
-      [class.ddp-required-question-prompt]="this.block.isRequired"
-      [innerHTML]="block.confirmPrompt"
-    ></p>
+        <input
+            matInput
+            type="text"
+            [formControl]="formGroup.get(controlName)"
+            [minlength]="block.minLength"
+            [maxlength]="block.maxLength"
+            [pattern]="block.regexPattern"
+            [placeholder]="placeholder || block.placeholder"
+            [attr.data-ddp-test]="'answer:' + block.stableId"
+            [matAutocomplete]="autoCompleteFromSource"
+            (input)="onInput($event.target.value)"
+        />
 
-    <mat-form-field class="example-input-field-confirmation">
-      <input
-        matInput
-        type="text"
-        [formControl]="formGroup.get(confirmationControlName)"
-        [placeholder]="block.confirmPlaceholder || placeholder"
-        [attr.data-ddp-test]="'answer-confirmation:' + block.stableId"
-      />
+        <mat-autocomplete
+            #autoCompleteFromSource="matAutocomplete"
+            class="autoCompletePanel"
+        >
+            <mat-option
+                *ngFor="let suggestion of filteredSuggestions$ | async"
+                class="autoCompleteOption"
+                [value]="suggestion.value"
+            >
+                <span
+                    class="suggestion"
+                    [innerHTML]="
+                        generateSuggestionOptionInnerHtml(
+                            suggestion,
+                            'span',
+                            'suggestion-match'
+                        )
+                    "
+                ></span>
+            </mat-option>
+        </mat-autocomplete>
     </mat-form-field>
-  </ng-container>
+
+    <ng-container *ngIf="block.confirmEntry">
+        <p
+            *ngIf="block.confirmPrompt"
+            class="ddp-question-prompt ddp-question-prompt-confirmation"
+            [class.ddp-required-question-prompt]="this.block.isRequired"
+            [innerHTML]="block.confirmPrompt"
+        ></p>
+
+        <mat-form-field class="example-input-field-confirmation">
+            <input
+                matInput
+                type="text"
+                [formControl]="formGroup.get(confirmationControlName)"
+                [placeholder]="block.confirmPlaceholder || placeholder"
+                [attr.data-ddp-test]="'answer-confirmation:' + block.stableId"
+            />
+        </mat-form-field>
+    </ng-container>
+    <p class="ddp-helper">
+        <span class="ddp-counter-color">
+            {{
+                charactersRemaining ?? block.answer
+                    | charactersRemaining: block.maxLength
```

Comment:
Not seeing that using the pipe is an advantage here. I mean, generally pipes are a way to achieve reusability, but I think here it just makes things less clear.